src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,151 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private int size;
+
+    public MyLinkedList() {
+        head = null;
+        size = 0;
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (head == null) {
+            head = newNode;
+        } else {
+            Node<T> current = head;
+            while (current.next != null) {
+                current = current.next;
+            }
+            current.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+
+        Node<T> newNode = new Node<>(value);
+        if (index == 0) {
+            newNode.next = head;
+            head = newNode;
+        } else {
+            Node<T> prevNode = getNode(index - 1);
+            newNode.next = prevNode.next;
+            prevNode.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        isNotValidIndex(index);
+
+        Node<T> current = getNode(index);
+        return current.element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        isNotValidIndex(index);
+
+        Node<T> current = getNode(index);
+        T oldValue = current.element;
+        current.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
+
+        Node<T> removedNode;
+        if (index == 0) {
+            removedNode = head;
+            head = head.next;
+        } else {
+            Node<T> prevNode = getNode(index - 1);
+            removedNode = prevNode.next;
+            prevNode.next = removedNode.next;
+        }
+        size--;
+        return removedNode.element;
     }
 
     @Override
     public boolean remove(T object) {
+        if (isEmpty()) {
+            return false;
+        }
+
+        Node<T> current = head;
+        Node<T> prev = null;
+
+        while (current != null) {
+            if (Objects.equals(current.element, object)) {
+                if (prev == null) {
+                    head = current.next;
+                } else {
+                    prev.next = current.next;
+                }
+                size--;
+                return true;
+            }
+            prev = current;
+            current = current.next;
+        }
+
```

Comment:
You should create a new method unlink as was mentioned in common mistakes.
![image](https://github.com/mate-academy/jv-linked-list/assets/83081198/9fd9e171-35c5-4134-b212-7da5814ed812)